CONTRIBUTING.md

```diff
@@ -61,6 +61,39 @@ This basically builds `terracognita` with the current sources.
 You also need to install other code dependencies not mandatory in the runtime environment:
   * [enumer](https://github.com/dmarkham/enumer) is used to generate some code.
   * [goimports](https://godoc.org/golang.org/x/tools/cmd/goimports) is used to format / organize code and imports. CI will perform a check on this, we highly recommend to run `$ goimports -w <your-modified-files>`.
+  * [mockgen](https://github.com/golang/mock) is used to generates the mocks for testing
+
+#### Add a new Provider
+
+:warning: It has to support `github.com/hashicorp/terraform-plugin-sdk@v2.0.0` or higher, providers with `v1` will not work anymore :warning:
+
+All the provider names of packages and so one are the ones used by the provider itself, the one in which all the resources are prefixed with (ex: `aws_instance` is `aws`)
+
+**Provider**
+
+All providers have to follow the [provider.Provider](https://github.com/cycloidio/terracognita/blob/master/provider/provider.go#L14) interface in order to be able to be integrated to the codebase. The best way would be to follow any of the other providers and modify it accordingly to your needs. Also reading the documentation of the interface will help a lot in any question regarding what means any of the functions.
+
+New providers have to be on the root with the package name being the name of the provider on Terraform (ex: Amazon is `aws`)
+
+The `NewProvider` will need all the information required to initialize the specific provider client(sdk), and the specific provider configuration so then Terraform can use it to read information remotely and load it to the Schemas. All the information has to be sent via parameters, no ENV variables.
+
+**Resources**
+
```

Comment:
I'd had a short notice such as: `In the following paragraphs, we will use AWS as example as it is the most advanced of the project, but the logic could apply similarly across the other providers too.`